source/MaterialXCore/Definition.cpp

```diff
@@ -93,15 +93,7 @@ vector<ShaderRefPtr> NodeDef::getInstantiatingShaderRefs() const
 bool NodeDef::validate(string* message) const
 {
     bool res = true;
-    validateRequire(hasType(), res, message, "Missing type");
-    if (isMultiOutputType())
-    {
-        validateRequire(getOutputCount() >= 2, res, message, "Multioutput nodedefs must have two or more output ports");
-    }
-    else
-    {
-        validateRequire(getOutputCount() == 0, res, message, "Only multioutput nodedefs support output ports");
-    }
+    validateRequire(!hasType(), res, message, "Nodedef should not have a type");
```

Comment:
Nitpick: maybe add to this message that instead of a type an explicit output should be used.